src/wattsi.pas

```diff
@@ -495,6 +495,23 @@    TCrossReferences = record
                if (LastSeenHeadingID = '') then
                   LastSeenHeadingID := 'blank-heading';
                LastSeenHeadingID := EnsureID(Element, LastSeenHeadingID).AsString;
+
+               // Append a self-link to each header
+               if (ClassName <> 'no-num no-toc') then
+               begin
+                  HeadingSelfLink := ConstructHTMLElement(eA);
+                  HeadingSelfLink.SetAttribute('class', 'self-link');
+
+                  Scratch := Default(Rope);
+                  Scratch.Append('#');
+                  ExtractedData := Element.GetAttribute('id');
+                  Assert(not ExtractedData.IsEmpty);
+                  Scratch.AppendDestructively(ExtractedData); // appending LastSeenHeadingID does not work
```

Comment:
@Hixie could you explain why simply appending LastSeenHeadingID does not work here? I don't understand how it can be different from GetAttribute('id'). Thanks!